scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,645 @@
+# Generalized simulated annealing implementation.
+# Copyright (c) 2016 Sylvain Gubian <sylvain.gubian@pmi.com>,
+# Yang Xiang <yang.xiang@pmi.com>
```

Comment:
Yang is a colleague of mine. We worked together on the R/C++ version and is happy with the BSD-like license.